### PR TITLE
Updates to codeowners

### PR DESCRIPTION
**Pull request by thomas-robinson**
**Description**
Updates the CODEOWNERS file

Fixes # 

**How Has This Been Tested?**
CI

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

**Comment by rem1776** [comment]
Should we try to fix the permissions so that it'll actually request reviews from everyone? We would just have to update Uriel's and Mikyung's permissions so they have write access.

**Review by rem1776 — APPROVED** [review]

**Review by bensonr — DISMISSED** [review]

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -52,18 +52,18 @@ cmake            @mlee03 @ngs333
...
-/test_fms/data_override/ @GFDL-Eric @rem1776
+#/data_override/ Currently no code owner
+/test_fms/data_override/ @rem1776
 
-/diag_manager            @thomas-robinson @ngs333
-/test_fms/diag_manager/  @thomas-robinson @ngs333
+/diag_manager            @thomas-robinson
+/test_fms/diag_manager/  @thomas-robinson
 
 /fv3gfs/          @bensonr
```
Comment (bensonr) [inline .github/CODEOWNERS:61]: I don't think we need this entry any more.

**Comment by thomas-robinson** [comment]
let's discuss at the meeting tomorrow

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -34,12 +34,12 @@ Makefile.am      @uramirez8707 @rem1776
...
-CM*              @mlee03 @ngs333
-cmake            @mlee03 @ngs333
+CM*              @mlee03
+cmake            @mlee03
 
 # Files specific to GitHub or GitLab
-/.github/        @GFDL-Eric @rem1776
-/.gitlab/        @GFDL-Eric @rem1776
+/.github/        @rem1776
+/.gitlab/        @rem1776
```
Comment (rem1776) [inline .github/CODEOWNERS:42]: we can get rid of this gitlab stuff too, this one and the yaml file below

**Review by bensonr — APPROVED** [review]

**Review by rem1776 — APPROVED** [review]

**Pull request merged**